### ISSUE TITLE
panic: 'byte index is not a char boundary'

### ISSUE THREAD

**Issue by mischnic** [parcel-bundler/source-map#69]
Non-ASCII characters aren't handled correctly:

```js
const SourceMap = require("@parcel/source-map").default;

const s = new SourceMap();
s.addSource("dist/rörfokus/4784.js")
console.log(s)
```

```
thread '<unnamed>' panicked at 'byte index 7 is not a char boundary; it is inside 'ö' (bytes 6..8) of `dist/rörfokus/4784.js`', parcel_sourcemap/src/utils.rs:59:60
```

https://github.com/parcel-bundler/source-map/blob/32401c9eb98abdd12f6aba21c6be2d18856901e7/parcel_sourcemap/src/utils.rs#L59-L61

This seems to be a better solution:
https://doc.rust-lang.org/std/string/struct.String.html#method.strip_prefix

**Comment by danieltroger** [comment]
Any updates on this?

**Comment by DeMoorJasper** [comment]
Oh didn't see this issue, will look into this

**Comment by danieltroger** [comment]
Thank you for the fix! I still have the issue in 2.0.0-nightly.776+ba7d2263, do I need to update source-map separately somehow?

**Comment by mischnic** [comment]
We'll need to release a new version of source-map and bump the version in Parcel itself.
